### PR TITLE
feat: Add IViewModel reference in factory methods to enable advanced …

### PR DESCRIPTION
**Pull request by jeanplevesque**
…implementations.

GitHub Issue: #

## Proposed Changes

- Feature

## What is the current behavior?

Advanced implementations of IDynamicProperty or IDynamicCommandStrategy can be difficult to make because there's no reference to the owning viewmodel.

## What is the new behavior?

`IDynamicPropertyFactory` and `IDynamicCommandBuilderFactory` now propagate an `IViewModel` owner when provided.

## Checklist

Please check if your PR fulfills the following requirements:

- [x] Documentation has been added/updated
- [x] Automated Unit / Integration tests for the changes have been added/updated
- [ ] Contains **NO** breaking changes
   **It contains breaking changes, but the version was bumped.**
- [ ] Updated the Changelog
- [ ] Associated with an issue

### CODE REVIEW

File: src/DynamicMvvm.Tests/Command/DynamicCommandBuilderFactoryTests.cs [hunk 1]
```diff
@@ -155,6 +156,16 @@ public async Task It_applies_strategies_in_order2()
 			testString.Should().Be("123");
 		}
 
+		[Fact]
+		public void It_passes_ViewModel_owner_correctly()
```
Comment (vincentcastagna) [inline src/DynamicMvvm.Tests/Command/DynamicCommandBuilderFactoryTests.cs:160]: Don't we wanna also test (adding a new test) if it does not pass, that the good exception is thrown ?

File: src/DynamicMvvm/Command/DynamicCommandBuilder.cs [hunk 2]
```diff
@@ -17,20 +17,25 @@ public class DynamicCommandBuilder : IDynamicCommandBuilder
...
+		/// <param name="viewModel">The <see cref="IViewModel"/> that will own the newly created <see cref="IDynamicCommand"/>.</param>
+		public DynamicCommandBuilder(string name, IDynamicCommandStrategy baseStrategy, IViewModel viewModel = null)
 		{
 			if (string.IsNullOrEmpty(name))
 			{
 				throw new ArgumentException($"'{nameof(name)}' cannot be null or empty", nameof(name));
 			}
 
 			Name = name;
+			ViewModel = viewModel;
```
Comment (vincentcastagna) [inline src/DynamicMvvm/Command/DynamicCommandBuilder.cs:29]: No issue if viewModel is actually null here for future usage ?

**Review by MatFillion — APPROVED** [review]

**Review by vincentcastagna — APPROVED** [review]

File: src/DynamicMvvm.Tests/Command/DynamicCommandBuilderFactoryTests.cs (hunk 1 above)
Comment (jeanplevesque) [reply src/DynamicMvvm.Tests/Command/DynamicCommandBuilderFactoryTests.cs:160]: null is a correct value. I added a test that validates that :)

File: src/DynamicMvvm/Command/DynamicCommandBuilder.cs (hunk 2 above)
Comment (jeanplevesque) [reply src/DynamicMvvm/Command/DynamicCommandBuilder.cs:29]: No issue if null. I added more xml doc (in the interface) about it.

**Pull request merged**